### PR TITLE
refactor(frontend): make AgentChat caller-driven

### PR DESCRIPTION
**Pull request by Maxsky5**
## Summary

- Refactors the shared `AgentChat` surface into a caller-driven, task-agnostic UI module.
- Moves workflow scope, runtime readiness, and session-history policy into Agent Studio and read-only transcript adapters.
- Passes opaque transcript targets and runtime presentation data through the shared render tree.
- Preserves loading, blocked, retry, and detailed history-failure behavior.
- Keeps repository-chat routing, runtime contracts, persistence, and UI redesign out of scope.

## Goal

The shared chat module depended on Agent Studio task state, repository readiness, and runtime-definition context. This change makes the module reusable by requiring each caller to provide its transcript presentation, interaction policy, runtime display data, capabilities, and actions.

## Technical choices

- Models the rendered transcript as an explicit `AgentChatTranscriptPresentation` union with an opaque caller-owned target.
- Keeps readiness and history-failure notice derivation in a focused feature helper, including retry actions and diagnostic details.
- Resolves runtime tool presentation above message cards so deep render components do not read runtime context.
- Keeps workflow session-scope construction in Agent Studio and history loading in the read-only transcript adapter.
- Splits large message-card tests into focused harness, system-message, and tool-message suites.

## Testing

- [x] `bun run lint`
- [x] `bun run typecheck`
- [x] `bun run test`
- [x] `bun run build`
- [x] Cargo checks are not applicable because this repository has no `Cargo.toml` or Rust source.

## Checklist

- [x] No documentation update is required because public workflows, runtime contracts, persisted data, and setup do not change.
- [x] I added or updated relevant tests.
- [x] I kept failures explicit instead of adding fallback logic.
- [x] I linked the related task below.

## Related Issues

- OpenDucktor task: `openduckto-amqn`

## Breaking Changes

- [x] No breaking changes to public runtime or persisted contracts.

## Additional Context

The task QA verdict is approved. The branch is rebased onto the current `origin/main`.

## Summary by CodeRabbit

* **New Features**
  * Improved agent chat transcript handling, including clearer loading, failure, and empty states.
  * Added runtime-aware tool presentation with consistent names, workflow classification, approval outcomes, and status details.
  * Added blocked-runtime recheck support and improved interaction availability indicators.

* **Bug Fixes**
  * Improved rendering for tool calls, system messages, transcripts, errors, prompts, todos, and attachments.

* **Tests**
  * Expanded coverage for transcript states, runtime behavior, tool presentations, system messages, and chat interactions.

### CODE REVIEW

File: packages/frontend/src/components/features/agents/agent-chat/readonly-transcript/use-session-transcript-surface-model.ts [hunk 1]
```diff
@@ -79,15 +83,55 @@ export function useSessionTranscriptSurfaceModel({
...
+      disabled: runtimeReadiness.isLoadingChecks,
+      isPending: runtimeReadiness.isLoadingChecks,
+    }),
+    [runtimeReadiness.isLoadingChecks, runtimeReadiness.refreshChecks],
+  );
+  const chatReadiness = deriveAgentChatReadiness({
+    transcriptState: sessionHistory.transcriptState,
+    runtimeReadiness,
+    runtimeBlockedAction,
+  });
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/components/features/agents/agent-chat/readonly-transcript/use-session-transcript-surface-model.ts]: Fixed in e296afb206. The history hook now exposes a retry callback and pending state. The read-only caller passes that action to readiness, and an integration test proves that Retry reloads a failed transcript.

**Comment by Maxsky5** [comment]
Review cleanup is in e296afb206.

Fixed: read-only history retry, explicit blocked-runtime errors, session-loading-only window reset, semantic workflow-scope comparison, stable readiness memoization, complete public type exports, transcriptTarget naming, missing readiness and runtime-presentation coverage, the five CodeRabbit test cleanups, and dead read-only session factories.

Covered: tool display-name edge cases now live in the runtime-presentation tests.

Declined after source review: runtimePresentation reference equality is the required memo contract and both production callers memoize it; composer-less transcript views still need approval and question interactions; session-switch reset belongs to the window hook; the missing tool presentation throw is an explicit fail-fast invariant; the regular-tool fixture is an isolated test seam; fork parent identity is semantic row data; and the message-meta and 31-bit history version concerns are pre-existing changes that need separate design work.
